### DISCUSSION TITLE
What is the correct way to set the value of an anonymous struct property on the rust side?

### DISCUSSION THREAD

**Discussion by Ethanwolfie** [slint-ui/slint#2650] [General] [answered]
The setter method to set the anonymous struct takes an tuple as input, while the order of the tuple elements doesn't seem to correspond to the order of the anonymous struct that I defined in the slint file. How to correctly set the value of an anonymous struct?

**Comment by tronical** [comment]
Hm, where do you run into this anonymous struct?

**Reply by Ethanwolfie** [reply]
I wrote an application that calculates the statistics of an 16 bit image, and then sends back the result back for display. The code is something like this:
On the slint side: 
`in-out property<{min: float, max: float, mean: float}> stats;`
On the rust side:
`self.set_stats((max as f32, mean as f32, min as f32));`

**Reply by Ethanwolfie** [reply]
And also, show the current pixel value at the cursor:
On the slint side:
`in-out property<{x: int, y: int, val: float}> pos-and-val;`
On the rust side:
`self.set_pos_and_val((val as f32, x as i32, y as i32));`

**Comment by ogoffart** [comment, marked as answer]
I assume you mean something like this
```
export component Foo {
    property <{x: int, a: string}> foo;
}
```

Then is indeed an anonymous struct in the native language.  And the order is the **alphabetical order of the fields**. 
It's like that for technical reason, because the type information doesn't have the exact order (although we could fix that) but then it would be a breaking change.

I think the best would be to declare an actual struct:

```
export struct Xa { x: int, a: string }

export component Foo {
    property <Xa> foo;
}
```

Then you have a non-annonymous struct that should be easier to handle.

**Reply by Ethanwolfie** [reply]
Thanks for the answer! I think maybe the above information should be added to the [Anonymous Structures](https://slint-ui.com/releases/1.0.1/docs/slint/src/reference/types.html#anonymous-structures) section of the documentation in case of other people running into the same problem.

**Reply by ogoffart** [reply]
I've added some documentation in https://github.com/slint-ui/slint/pull/2659  
But it will be in the rust documentation.
